cdc/scheduler/internal/tp/capture_manager.go

```diff
@@ -0,0 +1,160 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tp
+
+import (
+	"github.com/pingcap/log"
+	"github.com/pingcap/tiflow/cdc/model"
+	"github.com/pingcap/tiflow/cdc/scheduler/internal/tp/schedulepb"
+	"go.uber.org/zap"
+)
+
+// CaptureState is the state of a capture.
+//
+//      ┌──────────────┐ Heartbeat Resp ┌─────────────┐
+//      │ Uninitialize ├───────────────>│ Initialized │
+//      └──────┬───────┘                └──────┬──────┘
+//             │                               │
+//  IsStopping │          ┌──────────┐         │ IsStopping
+//             └────────> │ Stopping │ <───────┘
+//                        └──────────┘
+type CaptureState int
+
+const (
+	// CaptureStateUninitialize means the capture status is unknown,
+	// no heartbeat response received yet.
+	CaptureStateUninitialize CaptureState = 1
+	// CaptureStateInitialized means owner has received heartbeat response.
+	CaptureStateInitialized CaptureState = 2
+	// CaptureStateStopping means the capture is removing, e.g., shutdown.
+	CaptureStateStopping CaptureState = 3
+)
+
+// CaptureStatus represent captrue's status.
+type CaptureStatus struct {
+	OwnerRev schedulepb.OwnerRevision
+	Epoch    schedulepb.ProcessorEpoch
+	State    CaptureState
+}
+
+func newCaptureStatus(rev schedulepb.OwnerRevision) *CaptureStatus {
+	return &CaptureStatus{OwnerRev: rev, State: CaptureStateUninitialize}
+}
+
+func (c *CaptureStatus) handleHeartbeatResponse(
+	resp *schedulepb.HeartbeatResponse, epoch schedulepb.ProcessorEpoch,
+) {
+	epochMismatch := c.State != CaptureStateUninitialize &&
+		c.Epoch.Epoch != epoch.Epoch
+	if epochMismatch {
+		log.Warn("tpscheduler: ignore heartbeat response",
+			zap.String("epoch", c.Epoch.Epoch),
+			zap.String("respEpoch", epoch.Epoch),
+			zap.Int64("ownerRev", c.OwnerRev.Revision))
+		return
+	}
+
+	if c.State == CaptureStateUninitialize {
+		c.Epoch = epoch
+		c.State = CaptureStateInitialized
+	}
+	if resp.IsStopping {
+		c.State = CaptureStateStopping
+	}
+}
+
+type captureManager struct {
+	OwnerRev schedulepb.OwnerRevision
+	Captures map[model.CaptureID]*CaptureStatus
+
+	// A logical clock counter, for heartbeat.
+	tickCounter   int
+	heartbeatTick int
+}
+
+func newCaptureManager(rev schedulepb.OwnerRevision, heartbeatTick int) *captureManager {
+	return &captureManager{
+		OwnerRev:      rev,
+		Captures:      make(map[string]*CaptureStatus),
+		heartbeatTick: heartbeatTick,
+	}
+}
+
+func (c *captureManager) captureTableSets() map[model.CaptureID]*CaptureStatus {
+	return c.Captures
+}
+
+func (c *captureManager) checkCaptureInitialized() bool {
+	allInitialized := true
+	for _, captrueStatus := range c.Captures {
+		if captrueStatus.State == CaptureStateUninitialize {
+			allInitialized = false
+		}
+	}
```

Comment:
this can be further simplified.

no need to check all captures before return `false`, and `allInitialized` is unnecessary.